File: algorithms/python/removeDuplicatesfromSortedArrayII/removeDuplicatesfromSortedArrayII.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/remove-duplicates-from-sorted-array-ii/
# Author: Miao Zhang
# Date:   2021-01-14

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i = 0
        for num in nums:
            if i < 2 or num != nums[i - 2]:
                nums[i] = num
                i += 1
        return i

File: algorithms/cpp/removeDuplicatesfromSortedArrayII/removeDuplicatesfromSortedArrayII.cpp
// Source: https://leetcode.com/problems/remove-duplicates-from-sorted-array-ii/
// Author: Miao Zhang
// Date:   2021-01-14

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int i = 0;
        for (auto num: nums) {
            if (i < 2 || num != nums[i - 2]) {
                nums[i] = num;
                i++;
            }
        }
        return i;
    }
};
